app/src/main/java/xdman/util/XDMUtils.java

```diff
@@ -50,23 +52,16 @@
 	private static final char[] invalid_chars = { '/', '\\', '"', '?', '*', '<',
 			'>', ':', '|' };
 
-	public static String decodeFileName(String str) {
+	public static String decodeFileName(String encoded) {
+		String str = URLDecoder.decode(encoded, StandardCharsets.UTF_8);

```

Comment:
Please use a fallback so that if URLDecoder throws an exception due to invalid url encoding, we still can decode some of the name using the old way. This happens when some websites uses broken url encoding

Reply:
Okay, no problem.